src/Generating/StructGen.cpp

```diff
@@ -50,14 +50,35 @@ void cStructGenTrees::GenFinish(cChunkDesc & a_ChunkDesc)
 				Dest = &a_ChunkDesc;
 			}
 
-			int NumTrees = GetNumTrees(BaseX, BaseZ, Dest->GetBiomeMap());
+			double NumTrees = GetNumTrees(BaseX, BaseZ, Dest->GetBiomeMap());
 
 			sSetBlockVector OutsideLogs, OutsideOther;
-			for (int i = 0; i < NumTrees; i++)
+			if (NumTrees < 1)
 			{
-				GenerateSingleTree(BaseX, BaseZ, i, *Dest, OutsideLogs, OutsideOther);
+				Vector3i Pos;
+				Pos.x = (m_Noise.IntNoise3DInt(BaseX + BaseZ, BaseZ, 0) / 19) % cChunkDef::Width;
+				Pos.z = (m_Noise.IntNoise3DInt(BaseX - BaseZ, 0, BaseZ) / 19) % cChunkDef::Width;
+
+				Pos.y = Dest->GetHeight(Pos.x, Pos.z);
+
+				// If this test is not done with exact position for some reason the trees are cut at Chunk border
```

Comment:
I'm concerned about this comment, `for some reason` means something is misunderstood, and that's always bad.

Reply:
@mBornand Can you explain what you mean by `exact position`? Do you mean the equality test (`<=` instead of `<`) or that you use block coords (`BaseX * cChunkDef::Width + Pos.x` instead of, for example, just `Pos.x`)? Or something else?

Reply:
It is just important that block position is used and not the coordinates of the chunk because trees can also generate across chunks.
I edited the comment.

Reply:
Ok, so it's about the absolute coords instead of relative. That is pretty much evident, so the comment is rather useless, I'd remove it altogether. As well as the empty lines between the z and xz assignments of Pos, in both branches.

Reply:
If the tree is on chunk border it is generated for both chunk, so if the coordinates change the tree is cut.

Reply:
Exactly, so the comment is not really needed, it is self-evident that full absolute coords must be used for the test.

Reply:
It is removed